### ISSUE TITLE
Drawer improvement

### ISSUE THREAD

**Issue by Klaranth** [pandorabox-io/in-game#116]
ThinkSome & sunkenhero
Drawer's should behave like chests on_dig and not drop all items

**Comment by S-S-X** [comment]
This is intended feature specifically built into drawers. It is unique feature among containers and in my opinion should not be changed.

Maybe integrate this issue / idea with #111 and add upgrade item that prevents digging drawers with items inside?

**Comment by Klaranth** [comment]
El_Naso ; SX ; chuche952 ; BuckarooBanzai ; Hedgehog
Make drawers a little less easy to break.

SX  
Allow crafting steel and stone drawers?

chuche952
Stone drawers sound weird to me. Maybe add stainless steel drawer as a hardened version? Sure harder to make than just steel or wood on techlevel

BuckarooBanzai 
I looked into that a while back: my problem was making good looking textures :/ Everything else in the drawers mod has an api and would be destined for that

Hedgehog 
For the stone texture , .... look into the textures of stone bloco's at NSSM.
Or the cobble chest texture?

Naso 
How about plastic drawers?

**Comment by Klaranth** [comment]
Huhhila
A more durable drawer would be nice - I mean one(s) with miss click-durability similar to those of jumpdrive:warp_device

**Comment by Klaranth** [comment]
El_Naso : 
Wrenched chests display their name in the player's inventory instead of "chest with items" unless no name is set.

Hedgehog:
Based on Naso's idea, is it possible to add a name to a drawer?  And if so, include that in Naso's suggestion for upgrade ?

El_Naso:
Could also display "drawer with [itemstring] x [quantity]" 

Skylord :
To prevent overflow in chest from generating items for players (eg, charcoal in Ecoport keeps overflowing since chests are a bit limited), I would suggest to add **shared drawers** (allowing players to grab items from drawers if the drawer is shared)

Hedgehog :
So, basically newer set of drawers. A whole new set of all types of drawers that can be shared?

El_Naso 
What about a drawer upgrade that allows it to be shared, much like placing a chest in a machine's upgrade slot?

**Comment by Klaranth** [comment]
stephen
Make drawers unbreakable while there is items init. It's really anoying when you accidently destroy one, and you got to pick up 50k blocks.

**Comment by SwissalpS** [comment]
> stephen Make drawers unbreakable while there is items init. It's really anoying when you accidently destroy one, and you got to pick up 50k blocks.

This has been declined several times, no need to repost it. See compromise propositions above.

**Comment by Klaranth** [comment]
Monniasza 
Extract items from drawers with filter injectors: this would allow easy autocrafting/storage system. The current solution requires carefully monitoring inputs, so there are enough items.
BuckarooBanzai
This would be rather simple but does not work for anything else than 1x1 drawers
Monniasza 
For that, a block like the Impulse Hopper from Ender IO would be useful.
Exo 
Should just use basic_machines:mover then. It's kinda like a wireless 2-way injector
It like moves items to a chest and you c an move it to ano machine like 5 blocks away...?

**Comment by SwissalpS** [comment]
> Extract items from drawers with filter injectors: this would allow easy autocrafting/storage system. The current solution requires carefully monitoring inputs, so there are enough items.

that is already provided by drawer controller. No matter if you use drawers or chests, you have to monitor in/out carefully anyway.
